pytorch_lightning/core/optimizer.py

```diff
@@ -102,11 +102,13 @@ def _on_trainer_init(self, trainer):
                 break
 
     @classmethod
-    def to_lightning_optimizer(cls, optimizer, trainer):
-        if isinstance(optimizer, LightningOptimizer):
-            return optimizer
-        optimizer = cls(optimizer)
-        optimizer._on_trainer_init(trainer)
+    def _to_lightning_optimizer(cls, optimizer, trainer, opt_idx):
```

Comment:
this is a breaking API change, need to add meta methods with deprecation warning...
- name chnage
- added required arg

Reply:
`_to_lightning_optimizer ` is only for internal use. It shouldn't break any API.

Reply:
it was a public api so you never know who was using it... so we shall be always careful what we make as public/protected...